backend/tests/test_course_build_processor.py

```diff
@@ -0,0 +1,478 @@
+                        "ReceiptHandle": "message1",
+                        "Body": json.dumps(mock_sqs_data),
+                    }
+                ]
+            },
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "MaxNumberOfMessages": 10,
+                "WaitTimeSeconds": 20,
+            },
+        )
+        sqs_stubber.add_response(
+            "delete_message",
+            {},
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "ReceiptHandle": "message1",
+            },
+        )
+
+        sqs_stubber.activate()
+        mocker_map = {"sqs": sqs_client}
+        mocker.patch("boto3.client", lambda client: mocker_map[client])
+        mocker.patch("sys.argv", [""])
+        course_build_processor.main()
+
+    assert exc_info.type == Exception
+
+
+def test_course_build_status_completed(mocker, db, create_course_builds):
+    course_builds = db.query(CourseBuild).all()
+
+    course_build = course_builds[2]
+
+    sqs_client = boto3.client("sqs", region_name="azeroth")
+    sqs_stubber = botocore.stub.Stubber(sqs_client)
+
+    mock_sqs_data = {"course_build_id": course_build.id}
+
+    mock_settings = mocker.Mock()
+    setattr(mock_settings, "SQS_QUEUE", "testqueue")
+    setattr(mock_settings, "SQS_POLL_INTERVAL_MINS", "1")
+    mocker.patch(
+        "rope.api.processors.course_build_processor.settings",
+        mock_settings,
+    )
+
+    engine = create_engine("postgresql://pguser:pgpassword@localhost/ropedb")
+    mock_factory = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+    mocker.patch(
+        "rope.api.processors.course_build_processor.get_db",
+        lambda: mock_factory,
+    )
+
+    sqs_stubber.add_response(
+        "get_queue_url",
+        {"QueueUrl": "https://testqueue"},
+        expected_params={"QueueName": "testqueue"},
+    )
+    sqs_stubber.add_response(
+        "receive_message",
+        {
+            "Messages": [
+                {
+                    "ReceiptHandle": "message1",
+                    "Body": json.dumps(mock_sqs_data),
+                }
+            ]
+        },
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "MaxNumberOfMessages": 10,
+            "WaitTimeSeconds": 20,
+        },
+    )
+    sqs_stubber.add_response(
+        "delete_message",
+        {},
+        expected_params={
+            "QueueUrl": "https://testqueue",
+            "ReceiptHandle": "message1",
+        },
+    )
+
+    sqs_stubber.activate()
+    mocker_map = {"sqs": sqs_client}
+    mocker.patch("boto3.client", lambda client: mocker_map[client])
+    mocker.patch("sys.argv", [""])
+    course_build_processor.main()
+
+    sqs_stubber.assert_no_pending_responses()
+
+
+def test_failed_course_build_missing_instructor_user_id(
+    mocker, db, create_course_builds
+):
+    with pytest.raises(Exception, match="id") as exc_info:
```

Comment:
Same comment as previous test cases about not writing the code / asserting on exceptions that are not handled. 

I'm actually not clear on the value of this test case. In practice if this were to happen I think the code would just hit a `KeyError` so it's not helping to address code coverage :man_shrugging: (or am I wrong?)

Reply:
The code would just hit a KeyError. Prior to make all of these updates, I think the code coverage report consider this test to cover an`Exception` caught where we use `logging.error(f"Failed to build course: {e}")`. There are a variety of things that could probably cause an exception to be raised in that `try`, but I tried to capture one potential exception (the `KeyError`) to at least confirm we'd hit that code path.

I've removed the test.